### PR TITLE
Add validation and model specs

### PR DESCRIPTION
**Pull request by benardop**
Changes I have made in this PR:
- Add the following validations:
      For the User model:
           -  Name must not be blank.
           -  PostsCounter must be an integer greater than or equal to zero.
     For the Post model:
           - Title must not be blank.
           - Title must not exceed 250 characters.
           -  CommentsCounter must be an integer greater than or equal to zero.
           - LikesCounter must be an integer greater than or equal to zero.
 -  Added unit specs for all of models' methods and validations.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      number_of_comments = first_post.comments_counter
+      expect(number_of_comments).to be 1
+    end
+
+    it 'updates comment counter equal to 2 for a given post' do
+      first_comment.update_comments_counter
+      second_comment = Comment.create(post: first_post, user: second_user, text: 'Hi Tom!')
+      second_comment.update_comments_counter
+      number_of_comments = first_post.comments_counter
+      expect(number_of_comments).to be 2
```
Comment (BertrandConxy) [inline spec/models/comment_spec.rb]: - [ ] Good job on writing tests 👏  However, they are all failing where they raise the `UnknownAttributeError`  unknown attribute --> `posts_counter` Here is the screenshot of the log I am getting 👇 
image

Please try to fix that error, I hope that they will be passing 👍

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    it 'returns 3 for three posts' do
+      second_post.save
+      third_post.save
+      fourth_post.save
+      posts = first_user.recent_3_posts
+      number_of_posts = first_user.recent_3_posts.count
+      texts = posts.pluck(:text)
+      expect(number_of_posts).to be 3
+      expect(texts).to eql [fourth_post.text, third_post.text, second_post.text]
```
Comment (BertrandConxy) [inline spec/models/user_spec.rb:60]: - [ ] Good job so far 👏 Kudos on writing the tests for custom methods (recent_3_posts)! in addition to that, you are expected to write the tests for the validations of each model.
For example here in the user model, as the name should be present, so 👇
```ruby
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    it 'updates post counter equal to 2 for a given user' do
+      first_post.update_posts_counter
+      second_post = Post.create(user: second_user, title: 'Hello', text: 'This is my second post', comments_counter: 2,
+                                likes_counter: 3)
+      second_post.update_posts_counter
+      number_of_posts = second_user.posts_counter
+      expect(number_of_posts).to be 2
+    end
+  end
```
Comment (BertrandConxy) [inline spec/models/post_spec.rb:101]: - [ ] Also, here in the posts_spec, you are expected to write tests for the validations. Please add them. I have given you the example of how to do it in the above comments 😃 👍

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @benardop, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 2 above)
Comment (benardop) [reply spec/models/user_spec.rb:60]: @BertrandConxy Thanks for your detailed Code Review feedback. Much appreciated.

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @benardop 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- test ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**